jwst/photom/photom.py

```diff
@@ -912,9 +924,9 @@ def save_area_nirspec(self, pix_area):
             # There is a slice_id column for selecting a matching slice, but
             # we're just going to average the pixel area for all slices.
             pixel_area = np.nanmean(pixarea)
-            slit.meta.photometry.pixelarea_arcsecsq = pixel_area
-            slit.meta.photometry.pixelarea_steradians = \
-                slit.meta.photometry.pixelarea_arcsecsq * A2_TO_SR
+            self.input.meta.photometry.pixelarea_arcsecsq = pixel_area
```

Comment:
Is the "input" model being updated in place throughout the step or is this the variable name chosen for the output model?

Reply:
Looks like self.input is a copy of the input datamodel, so isn't updated in place

Reply:
Yes, `self.input` is modified in-place.